visualdl/storage/binary_record.h

```diff
@@ -67,6 +67,9 @@ struct BinaryRecord {
   static int counter_;
 };
 
+// Initial static counter variable.
+int BinaryRecord::counter_ = 0;
```

Comment:
Are you sure its ok to define it in the .h file?  I think it might have a problem if you include this in a unit test and link the unit test with the compiled .so file.